ex03.exs

```diff
@@ -53,9 +53,32 @@ defmodule Ex03 do
   def pmap(collection, process_count, function) do
     # your code goes here
     # I'm hoping to see a simple pipeline as the body of this function...
+    collection
+    |> Ex03.splitbyprocess(process_count)
+    |> Ex03.mapfunctions(function)
+    |> List.flatten()
   end
 
-  # and here...
+  def splitbyprocess(collection, process_num) do
+    Enum.chunk_every(collection, div(Enum.count(collection),process_num))
+  end
+
+
+  def mapfunctions(chunks, function) do
+    me = self()
+    Enum.map(chunks, fn (chunk) ->
+      spawn(fn -> process_chunk(function, chunk, me) end) end)
+    |> Enum.map(fn (pid) ->
+      receive do
+        {^pid, value} ->
+          value end
+        end)
+  end
```

Comment:
This is definitely doing thew right thing. however, it feels that it's doing too much for a single function. I'd really like to have seen this as two: `process_chunks` and `collect_results` or something similar.

# -2